hudi-client/hudi-java-client/src/test/java/org/apache/hudi/hadoop/TestHoodieFileGroupReaderOnHive.java

```diff
@@ -0,0 +1,329 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class TestHoodieFileGroupReaderOnHive extends TestHoodieFileGroupReaderBase<ArrayWritable> {
+
+  @Override
+  @Disabled
+  public void testReadLogFilesOnlyInMergeOnReadTable(RecordMergeMode recordMergeMode, String logDataBlockFormat) throws Exception {}
+
+  private static final String PARTITION_COLUMN = "datestr";
+  private static JobConf baseJobConf;
+  private static HdfsTestService hdfsTestService;
+  private static HoodieStorage storage;
+  private static FileSystem fs;
+  private static StorageConfiguration<Configuration> storageConf;
+
+  @BeforeAll
+  public static void setUpClass() throws IOException, InterruptedException {
+    // Append is not supported in LocalFileSystem. HDFS needs to be setup.
+    hdfsTestService = new HdfsTestService();
+    fs = hdfsTestService.start(true).getFileSystem();
+    storageConf = HoodieTestUtils.getDefaultStorageConf();
+    baseJobConf = new JobConf(storageConf.unwrap());
+    baseJobConf.set(HoodieMemoryConfig.MAX_DFS_STREAM_BUFFER_SIZE.key(), String.valueOf(1024 * 1024));
+    fs.setConf(baseJobConf);
+    storage = new HoodieHadoopStorage(fs);
+  }
+
+  @AfterAll
+  public static void tearDownClass() throws IOException {
+    hdfsTestService.stop();
+    if (fs != null) {
+      fs.close();
+      storage.close();
+    }
+  }
+
+  @Override
+  public StorageConfiguration<?> getStorageConf() {
+    return storageConf;
+  }
+
+  @Override
+  public String getBasePath() {
+    return tempDir.toAbsolutePath() + "/myTable";
+  }
+
+  @Override
+  public HoodieReaderContext<ArrayWritable> getHoodieReaderContext(String tablePath, Schema avroSchema, StorageConfiguration<?> storageConf) {
+    HoodieFileGroupReaderBasedRecordReader.HiveReaderCreator readerCreator = (s, j) -> new MapredParquetInputFormat().getRecordReader(s, j, null);
+    HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(storageConf).setBasePath(tablePath).build();
+    String tableName = metaClient.getTableConfig().getTableName();
+    JobConf jobConf = new JobConf(storageConf.unwrapAs(Configuration.class));
+    setupJobconf(jobConf);
+    HoodieArrayWritableAvroUtils.initCacheForTable(tableName, avroSchema, jobConf);
+    return new HiveHoodieReaderContext(readerCreator, tableName, getRecordKeyField(metaClient),
+        getStoredPartitionFieldNames(new JobConf(storageConf.unwrapAs(Configuration.class)), avroSchema));
+
+  }
+
+  @Override
+  public String getRecordPayloadForMergeMode(RecordMergeMode mergeMode) {
+    switch (mergeMode) {
+      case EVENT_TIME_ORDERING:
+        return DefaultHoodieRecordPayload.class.getName();
+      case OVERWRITE_WITH_LATEST:
+        return OverwriteWithLatestAvroPayload.class.getName();
+      case CUSTOM:
+      default:
+        return CustomPayloadForTesting.class.getName();
+    }
+  }
```

Comment:
Could we get rid of this?  The payload should be automatically inferred if needed.

Reply:
https://issues.apache.org/jira/browse/HUDI-8017